### PR TITLE
Refactor the last sets of widget definitions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/openrct2-ui/windows/MapGen.cpp [hunk 1]
```diff
@@ -116,58 +116,58 @@ static constexpr const int32_t WH = 273;
...
-    { WWT_RESIZE,   1,  0,          WW - 1, 43, WH - 2, 0xFFFFFFFF,                 STR_NONE },             /* WIDX_PAGE_BACKGROUND */ \
-    { WWT_TAB,      1,  3,          33,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_1 */ \
-    { WWT_TAB,      1,  34,         64,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_2 */ \
-    { WWT_TAB,      1,  65,         95,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_3 */ \
-    { WWT_TAB,      1,  96,         126,    17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE }              /* WIDX_TAB_4 */
+    MakeWidget     ({ 0, 43}, {250, 229}, WWT_RESIZE, 1, 0xFFFFFFFF, STR_NONE), /* WIDX_PAGE_BACKGROUND */ \
+    MakeRemapWidget({ 3, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB,    STR_NONE), /* WIDX_TAB_1 */ \
+    MakeRemapWidget({34, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB,    STR_NONE), /* WIDX_TAB_2 */ \
+    MakeRemapWidget({65, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB,    STR_NONE), /* WIDX_TAB_3 */ \
+    MakeRemapWidget({96, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB,    STR_NONE) /* WIDX_TAB_4 */
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/MapGen.cpp:123]:
```suggestion
    MakeWidget     ({ 0, 43}, {250, 229}, WWT_RESIZE, 1         ), /* WIDX_PAGE_BACKGROUND */ \
    MakeRemapWidget({ 3, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_1 */ \
    MakeRemapWidget({34, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_2 */ \
    MakeRemapWidget({65, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_3 */ \
    MakeRemapWidget({96, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB) /* WIDX_TAB_4 */
```

File: src/openrct2-ui/windows/MapGen.cpp [hunk 2]
```diff
@@ -116,58 +116,58 @@ static constexpr const int32_t WH = 273;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/MapGen.cpp:116]: Seems like the constants above went unused (?) If so, delete them

File: src/openrct2-ui/windows/Staff.cpp [hunk 3]
```diff
@@ -70,48 +70,41 @@ enum WINDOW_STAFF_WIDGET_IDX {
 validate_global_widx(WC_PEEP, WIDX_PATROL);
 validate_global_widx(WC_STAFF, WIDX_PICKUP);
 
+#define MAIN_STAFF_WIDGETS \
+    WINDOW_SHIM(WINDOW_TITLE, WW, WH), \
+    MakeWidget     ({      0,      43}, {190, 137}, WWT_RESIZE,        1                                        ), /* Resize */ \
+    MakeRemapWidget({      3,      17}, { 31,  27}, WWT_TAB,           1, SPR_TAB,        STR_STAFF_OVERVIEW_TIP), /* Tab 1 */ \
+    MakeRemapWidget({     34,      17}, { 31,  27}, WWT_TAB,           1, SPR_TAB,        STR_STAFF_OPTIONS_TIP ), /* Tab 2 */ \
+    MakeRemapWidget({     65,      17}, { 31,  27}, WWT_TAB,           1, SPR_TAB,        STR_STAFF_STATS_TIP   ), /* Tab 3 */ \
+    MakeRemapWidget({     96,      17}, { 31,  27}, WWT_TAB,           1, SPR_TAB                               )  /* Tab 4 */
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Staff.cpp:79]: Nice reuse :)

File: src/openrct2-ui/windows/Ride.cpp [hunk 4]
```diff
@@ -209,159 +209,159 @@ constexpr int32_t RCT1_LIGHT_OFFSET = 4;
...
+    MakeWidget        ({  7, 139}, {150, 12}, WWT_CHECKBOX, 1, STR_MINIMUM_WAITING_TIME,               STR_MINIMUM_LENGTH_BEFORE_DEPARTING_TIP     ),
+    MakeSpinnerWidgets({157, 139}, {152, 12}, WWT_SPINNER,  1, STR_ARG_10_STRINGID                                                                 ), // NB: 3 widgets
+    MakeWidget        ({  7, 154}, {150, 12}, WWT_CHECKBOX, 1, STR_MAXIMUM_WAITING_TIME,               STR_MAXIMUM_LENGTH_BEFORE_DEPARTING_TIP     ),
+    MakeSpinnerWidgets({157, 154}, {152, 12}, WWT_SPINNER,  1, STR_ARG_14_STRINGID                                                                 ), // NB: 3 widgets
+    MakeWidget        ({  7, 169}, {302, 12}, WWT_CHECKBOX, 1, STR_SYNCHRONISE_WITH_ADJACENT_STATIONS, STR_SYNCHRONISE_WITH_ADJACENT_STATIONS_TIP  ),
+    MakeWidget        ({ 21,  61}, {129, 12}, WWT_LABEL,    1                                                                                      ),
+    MakeWidget        ({ 21,  75}, {129, 12}, WWT_LABEL,    1, STR_LIFT_HILL_CHAIN_SPEED,              STR_NONE                                    ),
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Ride.cpp:269]:
```suggestion
    MakeWidget        ({ 21,  75}, {129, 12}, WWT_LABEL,    1, STR_LIFT_HILL_CHAIN_SPEED,                                                          ),
```

File: src/openrct2-ui/windows/MapGen.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/MapGen.cpp:123]: Nice catch!

File: src/openrct2-ui/windows/Ride.cpp (hunk 4 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Ride.cpp:269]: Good catch. Including the removal of the trailing comma, we get:
```suggestion
    MakeWidget        ({ 21,  75}, {129, 12}, WWT_LABEL,    1, STR_LIFT_HILL_CHAIN_SPEED                                                           ),
```

File: src/openrct2-ui/windows/MapGen.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/MapGen.cpp:116]: They're used in the `WINDOW_SHIM` macro, and elsewhere in the file to create the window.

File: src/openrct2-ui/windows/MapGen.cpp [hunk 5]
```diff
@@ -116,58 +116,58 @@ static constexpr const int32_t WH = 273;
 
 #define SHARED_WIDGETS \
     WINDOW_SHIM(WINDOW_TITLE, WW, WH), /* WIDX_BACKGROUND, WIDX_TITLE, WIDX_CLOSE */ \
-    { WWT_RESIZE,   1,  0,          WW - 1, 43, WH - 2, 0xFFFFFFFF,                 STR_NONE },             /* WIDX_PAGE_BACKGROUND */ \
-    { WWT_TAB,      1,  3,          33,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_1 */ \
-    { WWT_TAB,      1,  34,         64,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_2 */ \
-    { WWT_TAB,      1,  65,         95,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_3 */ \
-    { WWT_TAB,      1,  96,         126,    17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE }              /* WIDX_TAB_4 */
+    MakeWidget     ({ 0, 43}, {250, 229}, WWT_RESIZE, 1, 0xFFFFFFFF, STR_NONE), /* WIDX_PAGE_BACKGROUND */ \
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/MapGen.cpp:119]:
```suggestion
    MakeWidget     ({ 0, 43}, { WW, 229}, WWT_RESIZE, 1, 0xFFFFFFFF, STR_NONE), /* WIDX_PAGE_BACKGROUND */ \
```

File: src/openrct2-ui/windows/MapGen.cpp [hunk 6]
```diff
@@ -116,58 +116,58 @@ static constexpr const int32_t WH = 273;
...
-    { WWT_RESIZE,   1,  0,          WW - 1, 43, WH - 2, 0xFFFFFFFF,                 STR_NONE },             /* WIDX_PAGE_BACKGROUND */ \
-    { WWT_TAB,      1,  3,          33,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_1 */ \
-    { WWT_TAB,      1,  34,         64,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_2 */ \
-    { WWT_TAB,      1,  65,         95,     17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE },             /* WIDX_TAB_3 */ \
-    { WWT_TAB,      1,  96,         126,    17, 43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_NONE }              /* WIDX_TAB_4 */
+    MakeWidget     ({ 0, 43}, {250, 229}, WWT_RESIZE, 1         ), /* WIDX_PAGE_BACKGROUND */ \
+    MakeRemapWidget({ 3, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_1 */ \
+    MakeRemapWidget({34, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_2 */ \
+    MakeRemapWidget({65, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_3 */ \
+    MakeRemapWidget({96, 17}, { 31,  27}, WWT_TAB,    1, SPR_TAB) /* WIDX_TAB_4 */
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/MapGen.cpp:123]:
```suggestion
    MakeWidget     ({ 0, 43}, {WW, 229}, WWT_RESIZE, 1         ), /* WIDX_PAGE_BACKGROUND */ \
    MakeRemapWidget({ 3, 17}, {31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_1 */ \
    MakeRemapWidget({34, 17}, {31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_2 */ \
    MakeRemapWidget({65, 17}, {31,  27}, WWT_TAB,    1, SPR_TAB), /* WIDX_TAB_3 */ \
    MakeRemapWidget({96, 17}, {31,  27}, WWT_TAB,    1, SPR_TAB) /* WIDX_TAB_4 */
```

File: src/openrct2-ui/windows/Multiplayer.cpp [hunk 7]
```diff
@@ -58,14 +58,14 @@ enum WINDOW_MULTIPLAYER_WIDGET_IDX {
...
-    { WWT_TAB,              1,  3,      33,     17,     43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_PLAYERS_TIP },          /* tab                  */  \
-    { WWT_TAB,              1,  3,      33,     17,     43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_GROUPS_TIP },           /* tab                  */  \
-    { WWT_TAB,              1,  3,      33,     17,     43,     IMAGE_TYPE_REMAP | SPR_TAB,       STR_OPTIONS_TIP }           /* tab                  */  \
+    MakeWidget     ({  0,  0}, {340, 240}, WWT_FRAME,    0                                       ), /* panel / background */ \
+    MakeWidget     ({  1,  1}, {338,  14}, WWT_CAPTION,  0, STR_NONE,    STR_WINDOW_TITLE_TIP    ), /* title bar */ \
+    MakeWidget     ({327,  2}, { 11,  12}, WWT_CLOSEBOX, 0, STR_CLOSE_X, STR_CLOSE_WINDOW_TIP    ), /* close x button */ \
+    MakeWidget     ({  0, 43}, {340, 197}, WWT_RESIZE,   1                                       ), /* content panel */ \
+    MakeRemapWidget({  3, 17}, { 31,  27}, WWT_TAB,      1, SPR_TAB,     STR_SHOW_SERVER_INFO_TIP), /* tab */ \
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Multiplayer.cpp:65]: This looks odd. Why are all tabs sharing the same coordinates in this window?

File: src/openrct2-ui/windows/Network.cpp [hunk 8]
```diff
@@ -37,15 +37,12 @@ enum WINDOW_NETWORK_WIDGET_IDX {
...
-#define MAIN_NETWORK_WIDGETS \
-    { WWT_FRAME,            0,  0,          WW - 1,    0,      WH - 1,    STR_NONE,                   STR_NONE },                 /* panel / background   */  \
-    { WWT_CAPTION,          0,  1,          WW - 2,    1,      14,        STR_NONE,                   STR_WINDOW_TITLE_TIP },     /* title bar            */ \
-    { WWT_CLOSEBOX,         0,  WW - 13,    WW - 3,    2,      13,        STR_CLOSE_X,                STR_CLOSE_WINDOW_TIP },     /* close x button       */  \
-    { WWT_RESIZE,           1,  0,          WW - 1,    43,     WH - 1,    0xFFFFFFFF,                 STR_NONE },                 /* content panel        */  \
-    { WWT_TAB,              1,  3,          33,        17,     43,        IMAGE_TYPE_REMAP | SPR_TAB,       STR_SHOW_SERVER_INFO_TIP }, /* tab                  */  \
-
 static rct_widget window_network_information_widgets[] = {
-    MAIN_NETWORK_WIDGETS
+    MakeWidget     ({  0,  0}, {450, 210}, WWT_FRAME,    0                                       ), // panel / background
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Network.cpp:41]: WINDOW_SHIM ?

File: src/openrct2-ui/windows/Multiplayer.cpp (hunk 7 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Multiplayer.cpp:65]: Looks like the window leaves setting them up to `window_multiplayer_players_invalidate`, which leaves it up to `window_align_tabs`. We could also better initial values as well, though.

**Pull request merged**